src/stemmabench/algorithms/stemma.py

```diff
@@ -0,0 +1,209 @@
+            The following section is a suggestion of the possible contents of the dictionary.
+        #### Content:
+            - config (str, Optional): The path to the config file used to generate the tree.
+            - has_ground_truth (bool, Optional): Boolean indicating if the true tree is known. If false the path_to_original attribute will indicate the 
+            estimated original text.
+        """
+        self._set_folder_path(folder_path=folder_path)
+        self._edge_file: Union[str, None] = edge_file
+        self._generation_info: Union[Dict[str, Any], None] = generation_info
+        self._root: Union[ManuscriptBase, None] = None
+        self._text_lookup: Dict[str, ManuscriptBase] = {}
+        self._fitted: bool = False
+
+    @property
+    def root(self):
+        return self._root
+
+    @property
+    def folder_path(self):
+        return self._folder_path
+    
+    @property
+    def edge_file(self):
+        return self._edge_file
+    
+    @property
+    def fitted(self):
+        return self._fitted
+    
+    @property
+    def text_lookup(self):
+        return self._text_lookup
+    
+    @property
+    def generation_info(self):
+        return self._generation_info
+    
+    def _set_folder_path(self, folder_path: Union[str, None]) -> None:
+        """Checks that the folder path is an existing directory and sets the folder_path attribute.
+        
+        ### Args:
+            - folder_path (str): The path to the folder containing all the texts.
+
+        ### Raises:
+            - ValueError: If the specified folder_path is not an existing directory.
+        """
+        if folder_path and not os.path.isdir(folder_path):
+            raise ValueError("The folder_path specified is not an existing folder path.")
+        self._folder_path = folder_path
+    
+    def get_edges(self) -> None:
+        """Return array of all the edge values.
+        
+        ### Raises:
+            - NotImplementedError: Not implemented yet.
+        """
+        raise NotImplementedError()
+
+    def dict(self, include_edges: bool = False) -> Dict[str, dict]:
+        """Return a dict representation of the tree.
+        Dict is empty until tree is fitted (fitting can be done using .fit() method)
+
+        ### Args:
+            - include_edges (bool): Indicates if the dictionary should include
+
+        ### Returns:
+            - dict: A dictionay representaion of the calling stemma tree.
+
+        ### Raises:
+            - RuntimeError: If the stemma has not been fited yet.
+        """
+        if not self.fitted:
+            raise RuntimeError("Stemma not fitted yet.")
+        return self.root.dict(include_edges)
+    
+    def __eq__(self, value: object) -> bool:
+        """Returns true if both stemmas are the same. !!! Only checks that both stemmas contain all the same texts !!!
+        
```

Comment:
As you are unhappy with this method, you can add a TODO to improve it later.

Reply:
Added TODO